guardian/typerighter
# Upgrade dependencies to fix high vulnerabilities

## What does this change?

This upgrades dependencies to resolve [Snyk](https://app.snyk.io/org/guardian/project/8ecf100e-b5f7-43e5-bbaa-c28fcd28eb7a) high vulnerabilities. Some dependencies have been manually overriden as the libraries dependent on them were using versions with vulnerabilities.

The `sbt-dependency-graph` plugin has been added to ensure the Snyk CLI and dashboard are accurate. See more [here](https://support.snyk.io/hc/en-us/articles/360004167317-How-to-install-the-SBT-dependency-graph-plugin-to-test-Scala-projects-with-Snyk-CLI).

**Note**: there is one outstanding high vulnerability (`io.netty:netty-codec-http2`) introduced through the Guardian's `simple-configuration-ssm` package. There is an open [PR](https://github.com/guardian/simple-configuration/pull/63) addressing this. Once this is merged, the version of `simple-configuration-ssm` can be upgraded to resolve this.

## How to test

- Run the rule manager and checker apps locally, following README instructions. Do they run as expected?
- Deploy this branch to CODE and check that [rule manager](https://manager.typerighter.code.dev-gutools.co.uk/) and [checker](https://checker.typerighter.code.dev-gutools.co.uk/) run as expected

## How can we measure success?

There are no longer any high vulnerabilities (with the exception of the `io.netty:netty-codec-http2` vulnerability) in the Snyk [typerighter](https://app.snyk.io/org/guardian/project/8ecf100e-b5f7-43e5-bbaa-c28fcd28eb7a) project.
